### DISCUSSION TITLE
Rule syntax changed?

### DISCUSSION THREAD

**Discussion by Nonixnarrez** [arendst/Tasmota#19817] [Q&A]
Hello,

I´ve configured a Wemos D1 Mini Pro on Tasmota (13.2.0) with a cyble sensor for our gas meter (10k pull-up resistor on D5/GPIO14).
The sensor input GPIO port is configured as switch1 (switchmode 1 or 16) and the cyble sensor changes state from 1 to 0 each round of the gas meter. Messages can be seen in the Tasmota console window. So far so good.

Now I have configured a counter1 on another unused port D7/GPIO13 and wanted it to increase whenever switch1 state changes from 1 (ON) to 0 (OFF). The rule1 I´ve configured for that is: 
`rule1 on switch1#state=off do counter1 +1 endon`  ALTERNATIVELY `rule1 on switch1#state=0 do counter1 +1 endon` 
`rule1 5`
`rule1 1`

The counter can be seen in the main menu, however it does not increase with a state change reported from the switch.

Console output:
```
16:55:51.487 CMD: rule1
16:55:51.495 MQT: stat/sonoff/RESULT = {"Rule1":{"State":"ON","Once":"ON","StopOnError":"OFF","Length":41,"Free":470,"Rules":"ON Switch1#State=OFF DO Counter1 +1 ENDON"}}
16:56:15.382 MQT: tele/sonoff/STATE = {"Time":"2023-10-22T16:56:15","Uptime":"0T00:06:09","UptimeSec":369,"BatteryPercentage":0,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"RadiusX","BSSId":"B0:F2:08:BC:23:C9","Channel":11,"Mode":"11n","RSSI":56,"Signal":-72,"LinkCount":1,"Downtime":"0T00:00:03"}}
16:56:15.413 MQT: tele/sonoff/SENSOR = {"Time":"2023-10-22T16:56:15","Switch1":"OFF","COUNTER":{"C1":794586}}
16:57:15.407 MQT: tele/sonoff/STATE = {"Time":"2023-10-22T16:57:15","Uptime":"0T00:07:09","UptimeSec":429,"BatteryPercentage":0,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"RadiusX","BSSId":"B0:F2:08:BC:23:C9","Channel":11,"Mode":"11n","RSSI":52,"Signal":-74,"LinkCount":1,"Downtime":"0T00:00:03"}}
16:57:15.435 MQT: tele/sonoff/SENSOR = {"Time":"2023-10-22T16:57:15","Switch1":"OFF","COUNTER":{"C1":794586}}
16:58:15.408 MQT: tele/sonoff/STATE = {"Time":"2023-10-22T16:58:15","Uptime":"0T00:08:09","UptimeSec":489,"BatteryPercentage":0,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"RadiusX","BSSId":"B0:F2:08:BC:23:C9","Channel":11,"Mode":"11n","RSSI":50,"Signal":-75,"LinkCount":1,"Downtime":"0T00:00:03"}}
16:58:15.435 MQT: tele/sonoff/SENSOR = {"Time":"2023-10-22T16:58:15","Switch1":"OFF","COUNTER":{"C1":794586}}
``` 

I found older posts on the web, where people reported it to be working that way. Not sure if the code/syntax has changed ever since.

Thanks a lot for your support/guidance how to get it running in a stable way.

P.S.: I also tried to connect the cyble sensor directly to the counter1 GPIO, but reporting was never as accurate as desired, no matter what counterdebounce/counterdebouncelow/counterdebouncehigh settings I applied.

**Comment by barbudor** [comment]
You need to compare to 0 or 1, not off/on

**Reply by Nonixnarrez** [reply]
I´ve tried both ways and comparing to 0 instead also didn´t do the trick.

**Reply by sfromis** [reply]
Maybe you have set a [SwitchMode](https://tasmota.github.io/docs/Buttons-and-Switches/#switchmode) precluding you getting state 0 or 1? This includes the default mode 0 where you get a state value of 2 for toggle, instead of 0/1. Maybe `SwitchMode1 1` is what you want instead?
